src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp

```diff
@@ -0,0 +1,589 @@
+};
+
+static inline std::string
+GetKernelNameImplicitGemmV4R1Dynamic(const PerformanceImplicitGemmV4R1Dynamic& config,
+                                     AsmImplicitGemmKernelV4R1Fwd_t kernel_type)
+{
+    int GemmMRepeat = config.KPerBlock / (config.GemmMPerThreadSubC * config.GemmMLevel0Cluster *
+                                          config.GemmMLevel1Cluster);
+    int ThreadTileM = GemmMRepeat * config.GemmMPerThreadSubC;
+    int ThreadTileN = config.GemmNRepeat * config.GemmNPerThreadSubC;
+    std::string kernel_name_prefix;
+    if(AsmImplicitGemmV4R1 == kernel_type)
+        kernel_name_prefix = std::string("igemm_v4r1_dynamic_");
+    else // if (AsmImplicitGemmV4R1_1x1 == kernel_type)
+        kernel_name_prefix = std::string("igemm_v4r1_1x1_dynamic_");
+
+    return kernel_name_prefix + std::to_string(config.KPerBlock) + "x" +
+           std::to_string(config.BPerBlock * config.GemmNRepeat * config.GemmNPerThreadSubC) + "x" +
+           std::to_string(config.EPerBlock) + "_" + std::to_string(ThreadTileM) + "x" +
+           std::to_string(ThreadTileN) + "_" + std::to_string(config.GemmMPerThreadSubC) + "x" +
+           std::to_string(config.GemmMLevel0Cluster) + "x" +
+           std::to_string(config.GemmMLevel1Cluster) + "x" +
+           std::to_string(config.GemmNPerThreadSubC) + "x" +
+           std::to_string(config.GemmNLevel0Cluster) + "x" +
+           std::to_string(config.GemmNLevel1Cluster) + "_" +
+           std::to_string(config.InBlockCopyClusterLengths_E) + "x" +
+           std::to_string(config.InBlockCopyClusterLengths_N1) + "x" +
+           std::to_string(config.InBlockCopyClusterLengths_B) + "x" +
+           std::to_string(config.InBlockCopyClusterLengths_N2) + "_" +
+           std::to_string(config.WeiBlockCopyClusterLengths_E) + "x" +
+           std::to_string(config.WeiBlockCopyClusterLengths_K);
+}
+
+static inline int
+GetImplicitGemmV4R1DynamicBlockSize(const PerformanceImplicitGemmV4R1Dynamic& config)
+{
+    return config.GemmMLevel0Cluster * config.GemmNLevel0Cluster * config.GemmMLevel1Cluster *
+           config.GemmNLevel1Cluster;
+}
+
+static inline int
+GetImplicitGemmV4R1DynamicGridSize(const ConvolutionContext& ctx,
+                                   const PerformanceImplicitGemmV4R1Dynamic& config)
+{
+    const auto& N1 = config.GemmNRepeat;
+    const auto& N2 = config.GemmNPerThreadSubC;
+
+    const auto& n  = ctx.batch_sz;
+    const auto& k  = ctx.n_outputs;
+    const auto& ho = ctx.out_height;
+    const auto& wo = ctx.out_width;
+
+    const auto& b = (static_cast<std::size_t>(n) * ho * wo) / (static_cast<std::size_t>(N1) * N2);
+    const auto& b_per_block = config.BPerBlock;
+    const auto& k_per_block = config.KPerBlock;
+
+    return (b / b_per_block) * (k / k_per_block); // NOLINT
+}
+
+template <typename BotBufType, typename TopBufType, typename WeiBufType>
+static inline int RunAndMeasureSolutionDynamicBase(miopen::Handle& profile_h,
```

Comment:
@atamazov should we also stop using this function?

Reply:
Not yet. After #203, yes.

Reply:
Hi @asroy , @atamazov , I added a comment in front of `RunAndMeasureSolutionDynamicBase`, that this function need removed after #203 is OK. Currently just left it here